### PR TITLE
Milestone: connect API

### PR DESCRIPTION
**Pull request by Roseokpe**
In this milestone, I connected the React-Redux application to an existing API to create and remove books in a remote server.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -22,9 +29,55 @@ const booksReducer = (state = initialState, action) => {
...
+  const booksID = Object.keys(booksList);
+  const formatedBooks = [];
+  booksID.map((key) => formatedBooks.push({
+    id: key,
+    title: booksList[key][0].title,
+    author: booksList[key][0].author,
+    category: booksList[key][0].category,
+  }));
+  dispatch(fetchBook(formatedBooks));
+};
```
Comment (EricMbouwe) [inline src/redux/books/books.js]: - [ ] `createAsyncThunk` should be used for creating thunk action creators. It is the redux way to handle asynchronous actions.

eg: 
```js
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
);
```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -22,9 +29,55 @@ const booksReducer = (state = initialState, action) => {
...
+      title: newBook.title,
+      author: newBook.author,
+      category: newBook.category,
+    }),
+    headers: {
+      'content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(addBook(newBook));
+};
```
Comment (EricMbouwe) [inline src/redux/books/books.js]: - [ ] As mentioned above `createAsyncThunk` should be used for creating thunk action creators. It is the redux way to handle asynchronous actions.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -22,9 +29,55 @@ const booksReducer = (state = initialState, action) => {
...
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+    headers: {
+      'content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(removeBook(id));
+};
```
Comment (EricMbouwe) [inline src/redux/books/books.js]: - [ ] As mentioned above `createAsyncThunk` should be used for creating thunk action creators. It is the redux way to handle asynchronous actions.

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @Roseokpe, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- API connection ✔
- Create and delete book ✔
- Linters ✔
- PR ✔

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @Roseokpe, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**